lib/linalg.dx

```diff
@@ -148,21 +148,22 @@ def lu {n} (a: n=>n=>Float) :
     umat = upper_tri_mat uRef
     for j:n.
       for i:(..j).
-        i' = %inject i
+        i' = inject n i
         s = sum for k:(..i).
-          k'' = %inject k
-          k' = %inject k''
+          -- TODO: get transitivity of Subset working so we can do a single injection
+          k'' = inject (..j) k
+          k' = inject n k''
           ukj = get $ umat k' ((unsafe_nat_diff (ordinal j) (ordinal k))@_)
```

Comment:
Hopefully we can also tuck away these `unsafe_nat_diffs` into `inject` instances, but I'm not sure if that's straightforward because of the multiple indices needed to specify the injection.